index.js

```diff
@@ -233,7 +238,7 @@ NYC.prototype.walkAllFiles = function (dir, visitor) {
 }
 
 NYC.prototype._maybeInstrumentSource = function (code, filename, relFile) {
-  var instrument = this.exclude.shouldInstrument(filename, relFile)
+  var instrument = (!this.instrument && this.all) || this.exclude.shouldInstrument(filename, relFile)
```

Comment:
But users may be surprised/confused at the need to `include` their generated files, especially if the mechanism for doing so is conflated with the one for their source files. Anyway, for a "magic" workflow, `node_modules` should probably be blacklisted by default no matter what the fine-tuning options are.